dbms/src/Operators/Operator.h

```diff
@@ -0,0 +1,137 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Core/Block.h>
+
+#include <memory>
+
+namespace DB
+{
+/**
+ * All interfaces of the operator may return the following state.
+ * - finish status, spilling status and waiting status can be returned in all interfaces of operator.
+ * - operator may return a different running status depending on the interface.
+*/
+enum class OperatorStatus
+{
+    /// finish status
+    FINISHED,
+    CANCELLED,
+    /// spilling status
+    SPILLING,
+    /// waiting status
+    WAITING,
+    /// running status
+    // means that TransformOp/SinkOp needs to input a block to do the calculation,
+    // also used as an external state of `PipelineExec`.
+    NEED_INPUT,
+    // means that SourceOp/TransformOp output a block for used by subsequent operators.
+    // also used as an external state of `PipelineExec`.
+    HAS_OUTPUT,
+    // means that TransformOp handle the input block and output it.
+    PASS_THROUGH,
```

Comment:
I suggest changing this name for two reasons:
1. "Pass-through" sounds more like pass the data as is to the next operator without any processing;
2. It's an existing name in exchange mode.

Maybe "processing" or "computing" are better?

Reply:
how about removing PASS_THROUGH and using HAS_OUTPUT for transform_op.transform?
Because it does output the block after transform

Reply:
That sounds about right too.

Reply:
ok, updated.